File: top-interview-questions-easy/String/LeetCode125.go
package main

import (
	"fmt"
	"strings"
)

func isPalindrome(s string) bool {

	l, r := 0, len(s)-1

	if string(s[l]) != string(s[r]) {
		return false
	}

	for l < r {
		if s[l] == s[r] {
			l++
			r--
		} else {
			return false
		}
	}
	return true
}

func rightString(str string) string {
	str = strings.Replace(str, " ", "", -1)
	str = strings.Replace(str, ",", "", -1)
	str = strings.Replace(str, ":", "", -1)
	res := strings.ToLower(str)
	return res
}

func main() {
	str := "A man, a plan, a canal: Panama"
	res := rightString(str)
	fmt.Println(isPalindrome(res))
}

File: Offer/src/String/LeetCode125.java
package String;

public class LeetCode125 {

   static class Solution {
        public boolean isPalindrome(String s) {
            int left = 0, right = s.length()-1;
            while(left < right){
                while(left < right && !Character.isLetterOrDigit(s.charAt(left))){
                   left++;
               }
                while (left < right && !Character.isLetterOrDigit(s.charAt(right))){
                    right--;
                }

            }
            return true;
        }
    }

    public static void main(String[] args) {
            Solution solution = new Solution();
            boolean res =  solution.isPalindrome("A man, a plan, a canal: Panama");
            System.out.println(res);
    }
}
